server.js

```diff
@@ -1,30 +1,60 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
+import data from './data/books.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+//   Any port but 8080 is the default
 const port = process.env.PORT || 8080
 const app = express()
 
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Try /books')
+})
+
+// 20 result per page
+//books?page=2
+app.get('/books', (req, res) => {
+  let page = req.query.page ?? 1
+  const pageSize = 20
+  const maxPage = Math.ceil(data.length/pageSize)
+  page = (page<=maxPage && page>=1)?page : 1;
+  const startIndex = (page -1) * pageSize 
+  const endIndex = startIndex + pageSize
+  const resultPage = data.slice(startIndex, endIndex)
+  const result = {numBooks: resultPage.length, 
+  page : page,
+  maxPage: maxPage,
+  books: resultPage}
+  res.json(result)
+})
+
+app.get('/books/id/:id', (req, res) => {
+  const id = req.params.id
+  const bookId = data.find((item)=>item.bookID == id)
```

Comment:
You should use three plus signs when making checks like this, to make sure it's the same type as well. To make sure the id is a number (as we now the bookID is), you can set a plus sign in front of it to convert it to a number. If you want to read more, you can google "Unary plus".

`const bookId = data.find((item)=>item.bookID === +id)`

Reply:
And the variable should be named after what it contains, which in this case is the whole book object and not just the id. So maybe just call it `book` since it's returning a book :)